lib/spack/spack/test/concretize.py

```diff
@@ -417,6 +417,24 @@ def test_concretize_two_virtuals_with_dual_provider(self):
         """
         Spec('hypre ^openblas-with-lapack').concretize()
 
+    def test_concretize_propagate_disabled_variant(self):
+        """Test a package with a disabled propagated variant."""
+        Spec('hypre~~shared ^openblas').concretize()
+
+    def test_concretize_propagated_disabled_variant_passed_to_dependent(self):
```

Comment:
```suggestion
```

I think the second test is sufficiently simple that it's not worth separating the two.